### PR TITLE
Embed visual metrics into gecko profile

### PR DESCRIPTION
**Pull request by cliu55**
This change embeds visual metrics into gecko profiles captured by
browsertime, so they can be visualized in the Firefox profiler

### CODE REVIEW

File: lib/core/engine/iteration.js [hunk 1]
```diff
@@ -195,6 +198,21 @@ class Iteration {
...
+
+            const profileFilename = path.join(
+              this.engineDelegate.baseDir,
+              pathToFolder(result[i].url, options),
+              `geckoProfile-${index}.json`
+            );
+
+            try {
+              const geckoProfile = JSON.parse(fs.readFileSync(profileFilename));
+              geckoProfile.meta.visualMetrics = result[i].visualMetrics;
```
Comment (ncalexan) [inline lib/core/engine/iteration.js]: Can you make this work when `meta` is missing?  (Maybe it is never missing.)

File: lib/core/engine/iteration.js [hunk 2]
```diff
@@ -195,6 +198,21 @@ class Iteration {
               result[i].browserScripts.pageinfo.visualElements
             );
             result[i].visualMetrics = videoMetrics.visualMetrics;
+
+            const profileFilename = path.join(
```
Comment (ncalexan) [inline lib/core/engine/iteration.js]: Guard this with `browser === 'firefox'` whatever the flag for capturing Gecko profiles is, please.

File: lib/core/engine/iteration.js [hunk 3]
```diff
@@ -195,6 +198,21 @@ class Iteration {
...
+              pathToFolder(result[i].url, options),
+              `geckoProfile-${index}.json`
+            );
+
+            try {
+              const geckoProfile = JSON.parse(fs.readFileSync(profileFilename));
+              geckoProfile.meta.visualMetrics = result[i].visualMetrics;
+              fs.writeFileSync(profileFilename, JSON.stringify(geckoProfile));
+            } catch (e) {
+              log.error('Adding visual metrics to Gecko profile failed', e);
```
Comment (ncalexan) [inline lib/core/engine/iteration.js]: Be explicit about what was tried: include the filename in the error message, please.

**Review by ncalexan — CHANGES_REQUESTED** [review]
In general, I'm happy to take a commit like this, but I'd like to see more care around avoiding the IO and error if it's never going to work.

**Comment by cliu55** [comment]
Hi @ncalexan, the requested changes have been resolved.

**Review by ncalexan — APPROVED** [review]
Thanks!

**Pull request merged**